x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/TransportEqlSearchAction.java

```diff
@@ -125,25 +133,39 @@ public static void operation(PlanExecutor planExecutor, EqlSearchTask task, EqlS
         TimeValue timeout = TimeValue.timeValueSeconds(30);
         String clientId = null;
 
-        ParserParams params = new ParserParams(zoneId)
-            .fieldEventCategory(request.eventCategoryField())
-            .fieldTimestamp(request.timestampField())
-            .fieldTiebreaker(request.tiebreakerField())
-            .resultPosition("tail".equals(request.resultPosition()) ? Order.OrderDirection.DESC : Order.OrderDirection.ASC)
-            .size(request.size())
-            .fetchSize(request.fetchSize());
-
         RemoteClusterRegistry remoteClusterRegistry = new RemoteClusterRegistry(transportService.getRemoteClusterService(),
             request.indicesOptions());
-        EqlConfiguration cfg = new EqlConfiguration(request.indices(), zoneId, username, clusterName, filter,
+        Set<String> clusterAliases = remoteClusterRegistry.clusterAliases(request.indices(), false);
+        if (canMinimizeRountrips(request, clusterAliases)) {
+            String clusterAlias = clusterAliases.iterator().next();
```

Comment:
Instead of using two methods - `canMinimize` plus calling the cluster, how about combining them into a method that returns the cluster on which to re-route the call or otherwise returning null.
```
String singleCssCluster = minimizedCss(request, clusterAliases);
if (singleCssCluster != null) {
....
} else {
...
}
```

This helps with the different methods around checking, selecting and massaging the string into one method.

Reply:
I did initially consider this format too (before the small RemoteClusterRegistry refactoring), but found delegating the decision to a method a clearer implementation (similar to [TransportSearchAction#shouldMinimizeRoundtrips()](https://github.com/elastic/elasticsearch/blob/4f22f437ee50cacb94b37b457be1da0b8ba0e8ce/server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java#L332).